eva/catalog/models/index_io.py

```diff
@@ -0,0 +1,168 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from ast import literal_eval
+from typing import List
+
+from sqlalchemy import Boolean, Column, ForeignKey, Integer, String, UniqueConstraint
+from sqlalchemy.orm import relationship
+from sqlalchemy.types import Enum
+
+from eva.catalog.catalog_type import ColumnType, Dimension, NdArrayType
+from eva.catalog.models.base_model import BaseModel
+
+
+class IndexIO(BaseModel):
+    __tablename__ = "index_column"
+
+    _name = Column("name", String(100))
+    _type = Column("type", Enum(ColumnType), default=Enum)
+    _is_nullable = Column("is_nullable", Boolean, default=False)
+    _array_type = Column("array_type", Enum(NdArrayType), nullable=True)
+    _array_dimensions = Column("array_dimensions", String(100))
+
+    # Faiss index has both inputs and outputs as well. Both inputs and outputs
```

Comment:
Better if you move it to the top of the file.  Can you please document it as a schema? I cannot fully understand it. Thanks!

Reply:
Should I move to catalog index_io function or just top of file?